### PR TITLE
CCP-1078: committing for review

### PR DESCRIPTION
**Pull request by DannyShopgate**
## Description

Added terms and conditions to cart.item-list.after portal target and connect property to checkout button on cart.

All checkboxes must be checked in order to checkout. Did not add flag for required checked boxes because there isn't a requirement to do anything with that information at the moment

Used v6.1.0 of the Pwa to test.

### CODE REVIEW

File: extension-config.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+  "version": "1.0.0",
+  "id": "@shopgate/ext-terms-and-conditions",
```
Comment (sznowicki) [inline extension-config.json]: Id should be: @shopgate/terms-and-conditions

File: extension-config.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      "id": "isOrderableByTerms",
+      "path": "frontend/reducers/index.js",
+      "type": "reducers"
+    }
+  ],
+  "configuration": {
+    "checkBoxValues": {
+      "type": "admin",
+      "destination": "frontend",
+      "default": ["Value 1", "Value 2", "Value 3"],
```
Comment (sznowicki) [inline extension-config.json]: I'd suggest use different schema for the config so it's easy to extend it later:
[
{text: "Text that will be shown" }
]

Later we will extend it with optional/required config for each checkbox.

File: frontend/portals/ItemListAfter/index.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (sznowicki) [inline frontend/portals/ItemListAfter/index.jsx]: I'd suggest rename this file to CartItemListAfter so it's obvious that it's only rendered in the cart page

File: extension-config.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+  "version": "1.0.0",
+  "id": "@shopgate/ext-terms-and-conditions",
+  "components": [
+    {
+      "id": "TermsAndConditions",
+      "path": "frontend/portals/PaymentBarBefore",
```
Comment (sznowicki) [inline extension-config.json]: I failed to make this extension run. I did some small investigation and this path doesn't exist.
Please clean up the code and make sure it's ready for a review.

File: extension-config.json (hunk 1 above)
Comment (DannyShopgate) [reply extension-config.json]: Updated

File: extension-config.json (hunk 2 above)
Comment (DannyShopgate) [reply extension-config.json]: Updated to: 
        {
          "text": "Value 1",
          "textColor": "#000"
        }

File: frontend/portals/ItemListAfter/index.jsx (hunk 3 above)
Comment (DannyShopgate) [reply frontend/portals/ItemListAfter/index.jsx]: Renamed

File: extension-config.json (hunk 4 above)
Comment (DannyShopgate) [reply extension-config.json]: Ah my stupid mistake. I changed this before leaving yesterday.
Fixed.

File: frontend/components/CheckBoxes/index.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      return;
+    }
+    this.props.setCheckoutIsOrderable(true);
+  }
+
+  /**
+   * @returns {JSX}
+   */
+  render() {
+    let checkBoxes = null;
```
Comment (sznowicki) [inline frontend/components/CheckBoxes/index.jsx]: Why having null just to assign it to .map result?

**Comment by sznowicki** [comment]
Bug:
Cannot proceed to the checkout with default config and after checking all the checkboxes. Tested on iOS and browser.

File: frontend/components/CheckBoxes/style.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+const { colors } = themeConfig;
+
+const wrapper = css({
+  background: colors.shade8,
+}).toString();
+
+const content = css({
+  paddingTop: '15px',
+  width: '80%',
```
Comment (sznowicki) [inline frontend/components/CheckBoxes/style.js]: I'd suggest having 100% and variables.gap.big padding instead

File: frontend/components/CheckoutButton/index.jsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import { CHECKOUT_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
+import connect from './connector';
+import styles from './style';
+
+/**
+ * CheckoutButton renders button
+ * @param {Props} props props.
+ * @returns {JSX}
+ */
+const CheckoutButton = props => (
```
Comment (sznowicki) [inline frontend/components/CheckoutButton/index.jsx]: Did you maybe consider not implement custom checkout button but wrap the {props.children} with some wrapper that would add some opacity and prevent clicking the original one when checkout is blocked?

We could avoid maintenance of this when PWA6 one would change.

Just a suggesion.

Comment (dqdquint) [reply frontend/components/CheckoutButton/index.jsx]: Heyo, I didn't think about this before but I did just investigate incorporating context and I think its would be a great way to manipulate isActive with the connector prop isOrderable, but I'm not sure how to incorporate an new onClick function in this way in order to handle the notice mentioned above. I'm sure I must be missing something there.

Comment (DannyShopgate) [reply frontend/components/CheckoutButton/index.jsx]: *Was signed into wrong github account so deleted previous comment -_-*
Heyo, I didn't think about this before but I did just investigate incorporating context and I think its would be a great way to manipulate isActive with the connector prop isOrderable, but I'm not sure how to incorporate an new onClick function in this way in order to handle the notice mentioned above. I'm sure I must be missing something there.

**Comment by DannyShopgate** [comment]
It seems I adjusted extension name after testing and neglected to test again. Very sorry :(
Working on suggested updates.

**Comment by DannyShopgate** [comment]
Hey @sznowicki I looked into the implementation of toast and snackbar usage in the PWA, and I'm still not clear what would be the best design and implementation here. Any suggestions?

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+ The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
+
+ ### [1.0.0] - 2019-02-06
+First version of the extension.
+### Added
+- Added Terms and Conditions Checkboxes that must be checked in order to checkout.
+- You can configure an array of objects in the Shopgate Connect admin to display your desired text. Example config:
+```json
+{
+  "text": "Value 1", // Text to display
```
[the remark this answers is not in the document]
Comment (DannyShopgate) [reply CHANGELOG.md]: updated

File: frontend/components/CheckBoxes/index.jsx (hunk 5 above)
Comment (DannyShopgate) [reply frontend/components/CheckBoxes/index.jsx]: updated

File: frontend/components/CheckBoxes/style.js (hunk 6 above)
Comment (DannyShopgate) [reply frontend/components/CheckBoxes/style.js]: updated

File: frontend/components/CheckoutButton/index.jsx (hunk 7 above)
Comment (DannyShopgate) [reply frontend/components/CheckoutButton/index.jsx]: Please let me know if you have a better idea for this implementation. I pass the original checkout button when orderable.

**Comment by DannyShopgate** [comment]
Added Dialog modal to alert customers to check terms and conditions. Used the notice string from Lashbox LA website.

**Comment by sznowicki** [comment]
Just for the record (we talked off-line). I have one suggestion: make the default config empty, and if empty user can still checkout.

**Comment by sznowicki** [comment]
All looks good for now. Please just make it work with empty config and I think it can be merged then. Well done!

**Comment by DannyShopgate** [comment]
Sweet, thanks! Updated default state and tested.

**Pull request merged**